cmake/CheckSVEsupport.cmake

```diff
@@ -0,0 +1,32 @@
+function(check_xgboost_sve_support)
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+    include(CheckCSourceCompiles)
+
+    # Save the original C_FLAGS to restore later
+    set(ORIGINAL_C_FLAGS "${CMAKE_C_FLAGS}")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=armv8-a+sve")
```

Comment:
Rather than modifying `CMAKE_C_FLAGS` directly, we should use `CMAKE_REQUIRED_FLAGS` instead, which is [explicitly designed to influence the behavior of `check_c_source_compiles`](https://cmake.org/cmake/help/v3.31/module/CheckCSourceCompiles.html).

Example: https://github.com/facebook/rocksdb/blob/c0be6a4b90a1f616969b2a808035ebf334894a37/CMakeLists.txt#L309-L342

Let me update the pull request to use `CMAKE_REQUIRED_FLAGS`.